File: py/IEC62325/MarketOperations/MarketOpCommon/MarketInvoiceLineItem.py
from IEC61970.Base.Domain.DateTime import DateTime
from IEC61970.Base.Domain.DateTimeInterval import DateTimeInterval
from IEC62325.MarketOperations.MarketOpCommon.MarketInvoice import MarketInvoice
from IEC62325.MarketOperations.MktDomain.MktInvoiceLineItemKind import MktInvoiceLineItemKind
from IEC62325.MarketOperations.MarketSystem.MarketResults.Settlement import Settlement


class MarketInvoiceLineItem:
    """
    #  * An individual line item on an invoice.
    #  * @created 27-Dec-2023 5:14:58 PM
    """
    def __init__(self):
        self.bill_period = DateTimeInterval()  # Bill period for the line item.
        self.gl_account = ""  # General Ledger account code, shall be a valid combination.
        self.gl_date_time = DateTime()  # Date and time line item will be posted to the General Ledger.
        self.kind = MktInvoiceLineItemKind()  # Kind of line item.
        self.line_amount = 0.0  # Amount due for this line item.
        self.line_number = ""  # Line item number on invoice statement.
        self.line_version = ""  # Version number of the bill run.
        self.net_amount = 0.0  # Net line item charge amount.
        self.previous_amount = 0.0  # Previous line item charge amount.
        self.settlement = Settlement()
        self.container_market_invoice_line_item = MarketInvoiceLineItem()
        self.market_invoice = MarketInvoice()

File: java/IEC62325/MarketOperations/MarketOpCommon/MarketInvoiceLineItem.java
package IEC62325.MarketOperations.MarketOpCommon;

import IEC61970.Base.Domain.DateTimeInterval;
import IEC61970.Base.Domain.String;
import IEC61970.Base.Domain.DateTime;
import IEC62325.MarketOperations.MktDomain.MktInvoiceLineItemKind;
import IEC61970.Base.Domain.Float;
import IEC62325.MarketOperations.MarketSystem.MarketResults.Settlement;

/**
 * An individual line item on an invoice.
 * @created 03-Jan-2024 2:03:20 PM
 */
public class MarketInvoiceLineItem {

	/**
	 * Bill period for the line item.
	 */
	public DateTimeInterval billPeriod;
	/**
	 * General Ledger account code, shall be a valid combination.
	 */
	public String glAccount;
	/**
	 * Date and time line item will be posted to the General Ledger. 
	 */
	public DateTime glDateTime;
	/**
	 * Kind of line item.
	 */
	public MktInvoiceLineItemKind kind;
	/**
	 * Amount due for this line item.
	 */
	public Float lineAmount;
	/**
	 * Line item number on invoice statement.
	 */
	public String lineNumber;
	/**
	 * Version number of the bill run.
	 */
	public String lineVersion;
	/**
	 * Net line item charge amount.
	 */
	public Float netAmount;
	/**
	 * Previous line item charge amount. 
	 */
	public Float previousAmount;
	public Settlement Settlement;
	public MarketInvoiceLineItem ContainerMarketInvoiceLineItem;
	public MarketInvoice MarketInvoice;

	public MarketInvoiceLineItem(){

	}
}//end MarketInvoiceLineItem